### PR TITLE
feat: add support for variants update

### PR DESCRIPTION
**Pull request by djabarovgeorge**
### What change does this PR introduce?

Update the PUT /workflows endpoint to support the variants on the steps.
The parts of the code we have to update:
variables for all steps + variants
create message templates for variants
changes for variants

#### Definition of Done
the PUT /workflows endpoint supports the variants on the steps
e2e tests

### CODE REVIEW

File: apps/api/src/app/workflows/usecases/update-notification-template/update-notification-template.usecase.ts [hunk 1]
```diff
@@ -349,6 +363,65 @@ export class UpdateNotificationTemplate {
...
+  ): Promise<StepVariantEntity[]> {
+    if (!message.variants?.length) return [];
+
+    const variantsList: StepVariantEntity[] = [];
+    let parentVariantId: string | null = null;
+
+    for (const variant of message.variants) {
+      if (!variant.template) throw new ApiException(`Unexpected error: variants message template is missing`);
+
+      const variantTemplate = await this.createMessageTemplate.execute(
```
Comment (ainouzgali) [inline apps/api/src/app/workflows/usecases/update-notification-template/update-notification-template.usecase.ts:380]: It is possible that he is updating an existing variant, so it won't always be the createMessageTemplate @djabarovgeorge

File: apps/api/src/app/workflows/usecases/update-notification-template/update-notification-template.usecase.ts [hunk 2]
```diff
@@ -349,6 +363,65 @@ export class UpdateNotificationTemplate {
...
+        })
+      );
+
+      if (!variantTemplate._id) throw new ApiException(`Unexpected error: variants message template was not created`);
+
+      variantsList.push({
+        _id: variantTemplate._id,
+        _templateId: variantTemplate._id,
+        filters: variant.filters,
+        _parentId: parentStepId,
```
Comment (ainouzgali) [inline apps/api/src/app/workflows/usecases/update-notification-template/update-notification-template.usecase.ts:408]:
```suggestion
        _parentId: parentVariantId,
```

File: apps/api/src/app/workflows/usecases/update-notification-template/update-notification-template.usecase.ts (hunk 1 above)
Comment (djabarovgeorge) [reply apps/api/src/app/workflows/usecases/update-notification-template/update-notification-template.usecase.ts]: Will add the logic we have on the step update flow.

File: apps/api/src/app/workflows/e2e/update-notification-template.e2e.ts [hunk 3]
```diff
@@ -33,7 +33,13 @@ describe('Update workflow by id - /workflows/:workflowId (PUT)', async () => {
...
               template: {
                 type: StepTypeEnum.IN_APP,
-                content: 'This is new content for notification variant',
+                content: 'first content',
+              },
+            },
+            {
+              template: {
+                type: StepTypeEnum.IN_APP,
+                content: 'second content',
```
Comment (ainouzgali) [inline apps/api/src/app/workflows/e2e/update-notification-template.e2e.ts:42]: @djabarovgeorge just a note - it should not be possible to create/update a variant with no conditions

Comment (ainouzgali) [reply apps/api/src/app/workflows/e2e/update-notification-template.e2e.ts:42]: Updates or creates. Variants can not have empty conditions

**Pull request merged**